### ISSUE TITLE
Custom Input Field: Content Overlap and Auto-Scroll Issue When Keyboard Appears

### ISSUE THREAD

**Issue by TeHeal** [flyerhq/flutter_chat_ui#678]
In the case of a custom input field, there is an issue where the content gets covered. On the other hand, when the keyboard rises, I hope the messages can automatically scroll to the bottom.
自定义输入栏的情况下, 内容有覆盖的情况出现, 另一方面键盘抬起时, 希望能自动上拉消息到最低端
![screenshot_2024-11-16_12-45-34](https://github.com/user-attachments/assets/040ea786-468c-460c-bc1c-5c52f613b675)
![screenshot_2024-11-16_12-50-07](https://github.com/user-attachments/assets/f38291c3-93ff-452f-8e6c-ae2a5f39daef)

**Comment by demchenkoalex** [comment]
1. I will make `ChatInputHeightNotifier` available so you will be able to offset content to the height of the input, like I do in the example
2. I think you can supply `scrollController` to the chat already. You can use that to scroll to bottom when keyboard appears. I will see if I can add it inside automatically.

**Comment by wangsujin2015** [comment]
In this case, it's difficult to solve if the CustomScrollView's reverse property is not set to true

**Comment by demchenkoalex** [comment]
I will have to check that, will come back after I test

**Comment by zhangheng2022** [comment]
I have the same problem. ChatInputHeightNotifier  No export

**Comment by demchenkoalex** [comment]
example on how to use `ChatInputHeightNotifier` after I export it:

```dart
class CustomInput extends StatefulWidget {
  const CustomInput({super.key});

  @override
  State<CustomInput> createState() => _CustomInputState();
}

class _CustomInputState extends State<CustomInput> {
  final GlobalKey _inputKey = GlobalKey();

  @override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addPostFrameCallback((_) => _updateInputHeight());
  }

  @override
  Widget build(BuildContext context) {
    return Container(
      key: _inputKey,
      height: 50,
    );
  }

  void _updateInputHeight() {
    final renderBox =
        _inputKey.currentContext?.findRenderObject() as RenderBox?;
    if (renderBox != null) {
      context
          .read<ChatInputHeightNotifier>()
          .updateHeight(renderBox.size.height);
    }
  }
}
```

**Comment by demchenkoalex** [comment]
for the content moving up with a keyboard, this is the best I was able to achieve (delay is to wait for the final keyboard value to do a scroll animation otherwise Flutter sends like 20-30 smaller changes for the height and calling 20-30 jumps looks weird) I think this is better than nothing :)

https://github.com/user-attachments/assets/18ecbfcb-f1f5-4538-b75e-467642fc0ef2

**Comment by TeHeal** [comment]
Thank you Alex, we have received your seriousness! Come on together!
------------------------------------------------------------------
发件人：Alex ***@***.***>
发送时间：2024年12月28日(星期六) 07:07
***@***.***>
***@***.***>; ***@***.***>
主　题：Re: [flyerhq/flutter_chat_ui] Custom Input Field: Content Overlap and Auto-Scroll Issue When Keyboard Appears (Issue #678)
for the content moving up with a keyboard, this is the best I was able to achieve (delay is to wait for the final keyboard value to do a scroll animation otherwise Flutter sends like 20-30 smaller changes for the height and calling 20-30 jumps looks weird) I think this is better than nothing :)
https://github.com/user-attachments/assets/18ecbfcb-f1f5-4538-b75e-467642fc0ef2 <https://github.com/user-attachments/assets/18ecbfcb-f1f5-4538-b75e-467642fc0ef2 >

**Comment by demchenkoalex** [comment]
released in [v2.0.0-dev.2](https://github.com/flyerhq/flutter_chat_ui/releases/tag/2.0.0-dev.2)
